src/components/engagement/engagement.resolver.ts

```diff
@@ -44,7 +44,7 @@ export class EngagementResolver {
     @Args() { id, changeset }: ChangesetIds,
     @AnonSession() session: Session
   ): Promise<Engagement> {
-    const engagement = await this.service.readOne(id, session, changeset);
+    const engagement = await this.service.readOne(id, session, { changeset });
```

Comment:
How can we convert `ChangesetIds` to `ObjectView`?

Reply:
Hmm maybe a pipe in the decorator or a getter on the class. I could play with it quickly, shortly to see what feels best.

Reply:
I am thinking this should work with the `Transform` decorator... but it does not... 🧐
```
@ArgsType()
export class ChangesetIds {
  @IdField()
  id: ID;

  @IdField({ nullable: true, name: 'changeset' })
  @Transform(({ value }) => {
    const view: ObjectView = value ? { changeset: value } : { active: true };
    return view;
  })
  view: ObjectView;
}
```
Can you take a look at what I am missing?

Reply:
Mmm `@Transform` may not apply for `@ArgsType` since individual arguments are determined by graphql instead of an `@InputType` object sent straight through.

Reply:
Yeah... that makes sense... Do we have other options instead of `@Transform`?

Reply:
Use a Pipe transform. Example [here](https://github.com/SeedCompany/cord-api-v3/blob/develop/src/common/id.arg.ts#L13-L25), though it does something different.

Reply:
Hmm... can you write Pipe transform shortly? Not sure where I have to start... 😅

Reply:
Sure pushed a commit.